### DISCUSSION TITLE
Manual trigger for individual jobs

### DISCUSSION THREAD

**Discussion by elynnaie** [community/community#6211] [Actions]
**Feature Request**

I would like the ability to be able to manually trigger individual jobs in a workflow. The jobs would appear normally in the action run directed graph, but would be paused until a button was clicked in the UI.

**Use Case**

I have a variety of jobs that need to be run in a particular order, so they all need to exist in the same workflow (to make use of the `needs` syntax). However, I do not necessarily want them to run automatically. For example, I want to enforce that deployments happen first to the staging environment and then to production, but I don't want the deploy to production to happen immediately after the deploy to staging is finished.

Furthermore, there are some jobs that may not always need to be run, but if so, then they must be run at a particular point in the workflow. Having these jobs be on a manual trigger would allow me to skip them if they need to be skipped but still enforce that they run in the correct spot in the dependency tree.

**Comment by grossws** [comment]
There's [Environments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) for that. Unfortunately they are only available for Enterprise plan. 

A lot of people asked for manual approval support and it's available only via environments(

**Comment by ethomson** [comment]
As @grossws indicated, this is what [Environments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) are for.  Currently, environments are available for all public repositories, and for GitHub Enterprise customers.
